tekton-catalog/pipeline-loops/examples/loop-example-numeric-param.yaml

```diff
@@ -0,0 +1,80 @@
+apiVersion: custom.tekton.dev/v1alpha1
+kind: PipelineLoop
+metadata:
+  name: pipelineloop
+spec:
+  # IterationNumberParam points to the param that stores the current loop item. Similar to IterateNumeric
+  # for numeric loop range.
```

Comment:
`IterationNumberParam` points rather to collection or range index.

Reply:
I'd add the note that it's index in-order and that it starts from 1.